Source/Core/UICommon/ResourcePack/ResourcePack.cpp

```diff
@@ -0,0 +1,320 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "UICommon/ResourcePack/ResourcePack.h"
+
+#include <algorithm>
+
+#include <minizip/unzip.h>
+
+#include "Common/FileSearch.h"
+#include "Common/FileUtil.h"
+#include "Common/StringUtil.h"
+
+#include "UICommon/ResourcePack/Manager.h"
+#include "UICommon/ResourcePack/Manifest.h"
+
+namespace ResourcePack
+{
+ResourcePack::ResourcePack(const std::string& path) : m_path(path)
+{
+  auto file = unzOpen(path.c_str());
+
+  if (file == nullptr)
+  {
+    m_valid = false;
+    m_error = "Failed to open resource pack";
+    return;
+  }
+
+  if (unzLocateFile(file, "manifest.json", 0) == UNZ_END_OF_LIST_OF_FILE)
+  {
+    m_valid = false;
+    m_error = "Resource pack is missing a manifest.";
+    return;
+  }
+
+  unz_file_info manifest_info;
+
+  unzGetCurrentFileInfo(file, &manifest_info, nullptr, 0, nullptr, 0, nullptr, 0);
+
+  std::string manifest_contents;
+
+  manifest_contents.resize(manifest_info.uncompressed_size);
+
+  if (unzOpenCurrentFile(file) != UNZ_OK)
+  {
+    m_valid = false;
+    m_error = "Failed to open manifest.json";
+    return;
+  }
+
+  if (unzReadCurrentFile(file, &manifest_contents[0], manifest_info.uncompressed_size) <
+      static_cast<int>(manifest_contents.size()))
+  {
+    m_valid = false;
+    m_error = "Failed to read manifest.json";
+    return;
+  }
+
+  unzCloseCurrentFile(file);
+
+  m_manifest = std::make_shared<Manifest>(manifest_contents);
+
+  if (!m_manifest->IsValid())
+  {
+    m_valid = false;
+    m_error = "Manifest error: " + m_manifest->GetError();
+    return;
+  }
+
+  if (unzLocateFile(file, "logo.png", 0) != UNZ_END_OF_LIST_OF_FILE)
+  {
+    unz_file_info logo_info;
+
+    unzGetCurrentFileInfo(file, &logo_info, nullptr, 0, nullptr, 0, nullptr, 0);
+
+    m_logo_data.resize(logo_info.uncompressed_size);
+
+    if (unzOpenCurrentFile(file) != UNZ_OK)
+    {
+      m_valid = false;
+      m_error = "Failed to open logo.png";
+      return;
+    }
+
+    if (unzReadCurrentFile(file, &m_logo_data[0], logo_info.uncompressed_size) <
+        static_cast<int>(m_logo_data.size()))
+    {
+      m_valid = false;
+      m_error = "Failed to read logo.png";
+      return;
+    }
+
+    unzCloseCurrentFile(file);
+  }
+
+  unzGoToFirstFile(file);
+
+  do
+  {
+    std::string filename;
+
+    filename.resize(256);
+
+    unz_file_info texture_info;
+
+    unzGetCurrentFileInfo(file, &texture_info, &filename[0], sizeof(filename), nullptr, 0, nullptr,
+                          0);
+
+    if (filename.compare(0, 9, "textures/") != 0 || texture_info.uncompressed_size == 0)
+      continue;
+
+    // If a texture is compressed, abort.
```

Comment:
Still feels like it should be more of a guideline for the creator, not an error during installation.

Reply:
Probably the only meaningful way of enforcing this guideline is to refuse to load packs that violate it.

Reply:
Are you sure you can compare texture compression with data compression like that? But it probably is true it wouldn't bring much overall benefit as it wouldn't lower the filesize that much, including that it's simpler to implement a store-only VFS, and it would read or load all into memory (prefetch) faster than having to decompress the data live, so from that perspective I would actually support an enforcement, then the map makers have to be properly notified they need to pack into a store-only ZIP. 

@BhaaLseN If there is a case where size reduction is not bad then ofcourse, for example the high-quality raw PNGs, otherwise IMO the big purpose of ZIP is just the packaging into one file rather than thousands, it is probably faster on read/seek on HDDs as it's better for filesystem performance, versus a lot of small files.